lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -797,7 +797,7 @@ protected function prepare_item_for_database( $request ) {
 			}
 		}
 		// Post slug.
-		if ( ! empty( $schema['properties']['slug'] ) && isset( $request['slug'] ) ) {
+		if ( ! empty( $schema['properties']['slug'] ) && isset( $request['slug'] ) && ! empty( $request['slug'] ) && ! is_object( $request['slug'] ) ) {
```

Comment:
ps. I removed the is_object test and now my `test_update_draft_post_with_empty_slug` test fails